org.envirocar.app/src/org/envirocar/app/views/carselection/CarSelectionActivity.java

```diff
@@ -319,10 +319,19 @@ public void onCarAdded(Car car) {
             mCarListAdapter.addCarItem(car);
             showSnackbar(String.format(getString(R.string.car_selection_successfully_added_tmp),
                     car.getManufacturer(), car.getModel()));
+
+            // Check the total Cars count after adding, if only 1 car then set it.
+            int count = mCarListAdapter.getCount();
+            if(count == 1) {
+                mCarManager.setCar(car);
+                showSnackbar(String.format(getString(R.string.car_selection_car_selected_after_add),
+                        car.getManufacturer(), car.getModel()));
+            }
         } else {
             showSnackbar(String.format(getString(R.string.car_selection_already_in_list_tmp),
                     car.getManufacturer(), car.getModel()));
         }
+        setupListView();
```

Comment:
Instead of calling `setupListView()` we should adopt [addCarItem()](https://github.com/enviroCar/enviroCar-app/blob/5cd30ed7012b269f2dcd5f5f2ce195e5f39c2d1c/org.envirocar.app/src/org/envirocar/app/views/carselection/CarSelectionListAdapter.java#L207-L210) in a way that we set the selected car manually, e.g.:
```
protected void addCarItem(Car car) {
        this.mCars.add(car);
        if(this.getCount() == 1) {
            this.mSelectedCar = car;
        }
        notifyDataSetChanged();
    }
```
The call of `notifyDataSetChanged()` effects that the view is updated according to changes in the underlying dataset.